### DISCUSSION TITLE
program and icon from csv?

### DISCUSSION THREAD

**Discussion by prankousky** [davatorium/rofi#2019] [General]
Hi everybody,

I follow the example from the README.md and create those two files

`rofi_launcher.sh`

```bash
#!/bin/env bash

WORKINGDIR="/home/me/.local/share/bin"
MAP="$WORKINGDIR/rofi_programmstarter.csv"

cat "$MAP" \
    | cut -d ',' -f 1 \
    | rofi -dmenu -p "Util " \
    | head -n 1 \
    | xargs -i --no-run-if-empty grep "{}" "$MAP" \
    | cut -d ',' -f 2 \
    | head -n 1 \
    | xargs -i --no-run-if-empty /bin/bash -c "{}"

exit 0

```

`rofi_launcher.csv`

```csv
Firefox (def),firefox
Firefox (anderes Profil), firefox -p 123
```

This will work, as in it will give me two choices: 1. `Firefox (def)` 2. `Firefox (anderes Profil)`. So far, so good.

But would it be possible to add icons to the csv and get the script to parse them?  For example, change `rofi_launcher.csv` to this

```csv
Firefox (def),firefox,$HOME/.config/icons/firefox.png
Firefox (anderes Profil), firefox -p 123,/tmp/somepath/firefox2.png
```

But then, what would the `rofi_launcher.sh` have to look like?

I wasn't able to find a way to use custom icons, only icon themes, which (I assume) are based on the program names. So `firefox` should work, but `firefox123` should not work, because there is no default icon for this.

Thank you in advance for your ideas :)

**Comment by DaveDavenport** [comment]
yes, this is fully possible. you can pass an icon name, or a full path to an image.
See the manpages and examples in the repository .

**Reply by DaveDavenport** [reply]
```bash
echo -en "test\0icon\x1f${PWD}/current.jpg\n" | rofi -dmenu
```

**Reply by prankousky** [reply]
Thank you. I have tried this with multiple different files (png and jpg), but there is no image being displayed. It only shows "test", nothing else.

Are there restrictions, for example, maximum  file size / resolution? Or what could cause this? I am running `rofi 1.7.5`.

**Reply by DaveDavenport** [reply]
Did you enable icons in your theme?  This should work fine if there is a file 'current.jpg' in the folder you execute it from. (I will test on 1.7.5)

**Reply by DaveDavenport** [reply]
```bash
echo -en "test\0icon\x1f${PWD}/current.jpg\n" | rofi -dmenu -no-config -show-icons
```
this works for me with 1.7.5.

There is no file limit, it scales it on load.. the image I had as test is several megapixels big.

**Reply by prankousky** [reply]
Thank you. That works. I had missed the `-show-icons` parameter. :)
